In [89]:
import pandas as pd
import numpy as np

# Question 1:

Data1 File Operations:

In [90]:
# Reading file -data1
df_data1 = pd.read_excel("D:/Pranjali/POC/Chaitali/Data1.xlsx")

In [91]:
df_data1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9478 entries, 0 to 9477
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Order ID    9478 non-null   object
 1   Product ID  9478 non-null   object
 2   Qty         9478 non-null   int64 
dtypes: int64(1), object(2)
memory usage: 222.3+ KB


In [92]:
# Concatinating columns -col1, col2 for both datasets
df_data1['OrderID_ProductID_1'] = df_data1['Order ID']+df_data1['Product ID']
df_data1.head()

Order ID       Product ID  Qty            OrderID_ProductID_1
0  CA-2014-100006  TEC-PH-10002075    3  CA-2014-100006TEC-PH-10002075
1  CA-2014-100090  FUR-TA-10003715    3  CA-2014-100090FUR-TA-10003715
2  CA-2014-100090  OFF-BI-10001597    6  CA-2014-100090OFF-BI-10001597
3  CA-2014-100293  OFF-PA-10000176    6  CA-2014-100293OFF-PA-10000176
4  CA-2014-100328  OFF-BI-10000343    1  CA-2014-100328OFF-BI-10000343

Data2 File Operation :

In [93]:
df_data1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9478 entries, 0 to 9477
Data columns (total 4 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Order ID             9478 non-null   object
 1   Product ID           9478 non-null   object
 2   Qty                  9478 non-null   int64 
 3   OrderID_ProductID_1  9478 non-null   object
dtypes: int64(1), object(3)
memory usage: 296.3+ KB


In [96]:
# Reading file -data1
df_data2 = pd.read_excel("D:/Pranjali/POC/Chaitali/Data2.xlsx")

In [97]:
df_data2['OrderID_ProductID_2'] = df_data2['Order ID']+df_data2['Product ID']
df_data2.head(2)

Order ID       Product ID  Qty            OrderID_ProductID_2
0  CA-2014-100090  FUR-TA-10003715    3  CA-2014-100090FUR-TA-10003715
1  CA-2014-100090  OFF-BI-10001597    6  CA-2014-100090OFF-BI-10001597

In [101]:
df_data1['Match'] = df_data1['OrderID_ProductID_1'].isin(df_data2['OrderID_ProductID_2'])

In [102]:
df_data1_match = df_data1[['Order ID','Product ID','Qty','OrderID_ProductID_1','Match']]

In [103]:
df_data1_match['Match'] = df_data1_match['Match'].astype(str) 
print(df_data1_match)  

            Order ID       Product ID  Qty            OrderID_ProductID_1  \
0     CA-2014-100006  TEC-PH-10002075    3  CA-2014-100006TEC-PH-10002075   
1     CA-2014-100090  FUR-TA-10003715    3  CA-2014-100090FUR-TA-10003715   
2     CA-2014-100090  OFF-BI-10001597    6  CA-2014-100090OFF-BI-10001597   
3     CA-2014-100293  OFF-PA-10000176    6  CA-2014-100293OFF-PA-10000176   
4     CA-2014-100328  OFF-BI-10000343    1  CA-2014-100328OFF-BI-10000343   
...              ...              ...  ...                            ...   
9473  US-2017-169551  OFF-PA-10004100    3  US-2017-169551OFF-PA-10004100   
9474  US-2017-169551  OFF-ST-10004835    3  US-2017-169551OFF-ST-10004835   
9475  US-2017-169551  TEC-AC-10002018    3  US-2017-169551TEC-AC-10002018   
9476  US-2017-169551  TEC-AC-10003033    2  US-2017-169551TEC-AC-10003033   
9477  US-2017-169551  TEC-PH-10001363    2  US-2017-169551TEC-PH-10001363   

      Match  
0     False  
1      True  
2      True  
3      True  
4    

In [104]:
df_data1_match['Match'].value_counts()

True     8971
False     507
Name: Match, dtype: int64

In [105]:
df_data1_match.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9478 entries, 0 to 9477
Data columns (total 5 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Order ID             9478 non-null   object
 1   Product ID           9478 non-null   object
 2   Qty                  9478 non-null   int64 
 3   OrderID_ProductID_1  9478 non-null   object
 4   Match                9478 non-null   object
dtypes: int64(1), object(4)
memory usage: 370.4+ KB


In [106]:
df_data1_match = df_data1.to_csv("D:/Pranjali/POC/Chaitali/output_data1.csv", header=True)

Task1_Question1: How to identify the Records (Order ID + Product ID combination) present in data1 but missing in data2 (Specify the number of records missing in your answer)
Ans - 507 records are not matching from data1[OrderID_ProductID_1] with data2[OrderID_ProductID_2]     